### PR TITLE
Addon name is only unique for a given type (bug 1173609)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1173609](https://bugzilla.mozilla.org/show_bug.cgi?id=1173609)

The PR also completes/fixes PR #636

### CODE REVIEW

**Comment by kmaglione** [comment]
Looks good. You just have the types wrong in a couple of places (hence failing test). r+wc

File: apps/addons/tests/test_forms.py [hunk 1]
```diff
@@ -30,21 +31,45 @@ def setUp(self):
...
-        eq_(f.errors.get('name'), None)
-
-    def test_new_existing(self):
-        """
-        New add-ons shouldn't be able to use existing add-on names.
-        """
-        f = forms.AddonForm(dict(name=self.existing_name))
-        assert not f.is_valid()
-        eq_(f.errors['name'][0], self.error_msg)
+    def test_update_addon_non_existing_name(self):
```
Comment (magopian) [inline apps/addons/tests/test_forms.py]: The `AddonForm` is never used in the code to create new addons, it's only used to edit existing ones (from the API). I've thus updated those two tests, and added some new tests.

**Comment by magopian** [comment]
@kmaglione r? I believe I've fixed the issues.

**Comment by kmaglione** [comment]
r+

**Pull request merged**